src/track/beat.h

```diff
@@ -0,0 +1,126 @@
+#pragma once
+
+#include <QDebug>
+#include <QSharedPointer>
+
+#include "proto/beats.pb.h"
+#include "track/bpm.h"
+#include "track/frame.h"
+#include "track/timesignature.h"
+
+namespace mixxx {
+class Beat {
+  public:
+    enum Type {
+        BEAT,
+        DOWNBEAT
+    };
+    enum Marker {
+        NONE = 0,
+        TIME_SIGNATURE = 1 << 0,
+        BPM = 1 << 1
+    };
+    Q_DECLARE_FLAGS(Markers, Marker)
+    explicit Beat(FramePos framePos,
```

Comment:
Apparently, I cannot declare this constructor with constexpr since it includes `mixxx::TimeSignature` which in turn includes `track::io::TimeSignature` (generated by `protoc`) which is not constexpr. Am I missing something? Please let me know if I should read more on this somewhere.

Reply:
Ok, then this is not possible.